typescript-conditionals - Daniel Yount

TheDanYount opened the pull request:
![image](https://github.com/TheDanYount/c0724-code-solutions/assets/158097100/e36f819a-1662-4223-94d2-11d3ee3ed35b)

TheDanYount committed:
Completed the exercise

TheDanYount committed:
Completed the exercise

Whoops! Didn't realize I wrote the oldEnoughToDrink function with the oldEnoughToDrinkAndDrive's instructions! That's why it's always false!

brandon-moy commented on typescript-conditionals/main.ts:
```diff
@@ -0,0 +1,157 @@
+  return person.age >= 16;
+}
+
+function isOldEnoughToDrinkAndDrive(person: Person): boolean {
+  const theNum = 5;
+  if (theNum === 5 && typeof person !== 'string') {
+    return false;
+  }
+  return false;
+}
```
is a person old enough to drink or old enough to drive if a variable number is 5 and the type of the parameter is not a string?

brandon-moy reviewed (changes requested):
You created the function but your condition logic does not do what the function is named/supposed to do.

The function check if a person is old enough to drink and drive so we want to check if they are old enough to drink _and_ they are old enough to drive.

TheDanYount committed:
Completed the exercise

Made the isOldEnoughToDrinkAndDrive function work as if someone in an off mooment started to contemplate the question before they realized that the result is independent of conditionals.

brandon-moy reviewed (approved):
Exactly what I was looking for, great job 👍
